backend/src/crd/controller/scheduledworkflow/controller.go

```diff
@@ -457,65 +465,111 @@ func (c *Controller) submitNextWorkflowIfNeeded(ctx context.Context, swf *util.S
 			ScheduledWorkflow: swf.Name,
 		}).Infof("Submitting workflow for ScheduledWorkflow (%v): nothing to submit (next scheduled at: %v)",
 			swf.Name, commonutil.FormatTimeForLogging(nextScheduledEpoch))
-		return nil, nextScheduledEpoch, nil
+		return false, nextScheduledEpoch, nil
 	}
-
-	workflow, err = c.submitNewWorkflowIfNotAlreadySubmitted(ctx, swf, nextScheduledEpoch, nowEpoch)
+	var workflowName string
+	submitted, workflowName, err = c.submitNewWorkflowIfNotAlreadySubmitted(ctx, swf, nextScheduledEpoch, nowEpoch)
 	if err != nil {
 		log.WithFields(log.Fields{
 			ScheduledWorkflow: swf.Name,
 		}).Errorf("Submitting workflow for ScheduledWorkflow (%v): transient error while submitting workflow: %v",
 			swf.Name, err)
 		// There was an error submitting a new workflow.
 		// We should attempt to handle the schedule again at a later time.
-		return nil, nextScheduledEpoch, err
+		return false, nextScheduledEpoch, err
 	}
 	log.WithFields(log.Fields{
 		ScheduledWorkflow: swf.Name,
-		Workflow:          workflow.Get().Name,
+		Workflow:          workflowName,
 	}).Infof("Submitting workflow for ScheduledWorkflow (%v): workflow (%v) successfully submitted (scheduled at: %v)",
-		swf.Name, workflow.Get().Name, commonutil.FormatTimeForLogging(nextScheduledEpoch))
-	return workflow, nextScheduledEpoch, nil
+		swf.Name, workflowName, commonutil.FormatTimeForLogging(nextScheduledEpoch))
+	return submitted, nextScheduledEpoch, nil
 }
 
 func (c *Controller) submitNewWorkflowIfNotAlreadySubmitted(
 	ctx context.Context,
 	swf *util.ScheduledWorkflow, nextScheduledEpoch int64, nowEpoch int64) (
-	*commonutil.Workflow, error) {
+	bool, string, error) {
 
 	workflowName := swf.NextResourceName()
 
 	// Try to fetch this workflow
 	// If it already exists, it means that it was already created in a previous iteration
 	// of this controller but that the controller failed to save this data.
-	foundWorkflow, isNotFoundError, err := c.workflowClient.Get(swf.Namespace,
+	_, isNotFoundError, err := c.workflowClient.Get(swf.Namespace,
 		workflowName)
 	if err == nil {
 		// The workflow was already created by a previous iteration of this controller.
 		// Nothing to do except returning the information needed by the controller to update
 		// the ScheduledWorkflow status.
-		return foundWorkflow, nil
+		return true, workflowName, nil
 	}
 
 	if !isNotFoundError {
 		// There was an error while attempting to retrieve the workflow
-		return nil, err
+		return false, "", err
 	}
 
-	// If the workflow is not found, we need to create it.
-	newWorkflow, err := swf.NewWorkflow(nextScheduledEpoch, nowEpoch)
-	createdWorkflow, err := c.workflowClient.Create(ctx, swf.Namespace, newWorkflow)
+	if swf.Spec.PipelineID != nil {
+
+		log.WithFields(log.Fields{
+			ScheduledWorkflow: swf.Name,
+			Workflow:          workflowName,
+		}).Infof("PipelineID used when submitting run",
```

Comment:
@NikeNano  I think this is the root cause of the backend test failure. You didn't add formatting directive like "%v". You should be able to reproduce it when you run the backend test in local.

Reply:
Thank you @capri-xiyue approciate, this is obviously wrong but what did you run to catch it? For me, locally the tests seems to run fine, I cant figure out what I am missing ...

Reply:
I use the IDE `GOLAND` to run the backend tests, looks like `GOLAND` uses command `go test backend`.
Did you run the tests in local with go command line directly in terminal? I found the output of the go command line in terminal is not easy to read, sometimes I  ignore some errors. For me, reading the results of running tests in IDE  is more user friendly.

Reply:
Ok, thank you. I use the terminal to run the test, agree that it is really hard to read the outputs and scan for the failed test. I looked in the raw logs and could not find anything in them either. Will give `GOLAND` a test :)